### ISSUE TITLE
Keycloak OIDC - bad request when enabling the auth provider on a newer Keycloak version

### ISSUE THREAD

**Issue by izaac** [rancher/dashboard#3406]
Rancher version: `2.6 master-head (07/06/2021))` _3e44415_
Related issue: https://github.com/rancher/dashboard/issues/2191

**Steps to Reproduce**:

- Rancher HA RKE1 
- Enable Auth Provider Keycloak OIDC

**Result**:

*Request Payload included at the bottom of this issue description*
`{"POST":{"scheme":"https","host":"<REDACTED>","filename":"/v3/keyCloakOIDCConfigs/keycloakoidc","query":{"action":"testAndApply"},"remote":{"Address":"<REDACTED>"}}}`
```json
{
    "baseType": "error",
    "code": "ServerError",
    "message": "[generic oidc]: server error while authenticating: oauth2: cannot fetch token: 400 Bad Request\nResponse: {\"error\":\"invalid_grant\",\"error_description\":\"Code not valid\"}",
    "status": 500,
    "type": "error"
}
```

**Additional Details**:

Keycloak version: `13.0.0`
This works on Keycloak version: `3.3.0.Final`

![Screen Shot 2021-07-06 at 1 46 27 PM](https://user-images.githubusercontent.com/27659/124664630-98664d00-de60-11eb-8209-9d16d33e53ac.png)

**More info**:

Request payload:

```json
{
    "enabled": true,
    "oidcConfig": {
        "actions": {
            "configureTest": "https://<REDACTED>/v3/keyCloakOIDCConfigs/keycloakoidc?action=configureTest",
            "testAndApply": "https://<REDACTED>/v3/keyCloakOIDCConfigs/keycloakoidc?action=testAndApply"
        },
        "baseType": "authConfig",
        "created": "2021-07-06T16:44:33Z",
        "createdTS": 1625589873000,
        "creatorId": null,
        "enabled": true,
        "id": "keycloakoidc",
        "labels": {},
        "links": {
            "self": "https://<REDACTED>/v3/keyCloakOIDCConfigs/keycloakoidc",
            "update": "https://<REDACTED>/v3/keyCloakOIDCConfigs/keycloakoidc"
        },
        "type": "keyCloakOIDCConfig",
        "uuid": "<REDACTED>",
        "_name": "keycloakoidc",
        "accessMode": "unrestricted",
        "rancherUrl": "https://<REDACTED>/verify-auth",
        "scope": "profile,email",
        "issuer": "http://<REDACTED>/auth/realms/amp",
        "authEndpoint": "http://<REDACTED>/auth/realms/amp/protocol/openid-connect/auth",
        "clientSecret": "<REDACTED>",
        "clientId": "<REDACTED>"
    }
}
```

**Comment by richard-cox** [comment]
I'm working with @brendarearden to address this. It looks like the real error was being hidden, I've created #3435 to address that. The underlying issue is scope related and could be either ui, backend or keycloak config related

**Comment by richard-cox** [comment]
Frontend fixes have merged via rancher/dashboard#3434, just waiting on backend fixes via https://github.com/rancher/rancher/pull/33580

**Comment by izaac** [comment]
13.0.0 On Rancher version `2.6 master-head (07/14/2021)`   _8073489_

Sanity check of the Keycloak instance
3.3.0.Final On Rancher version `2.6 master-head (07/15/2021)`  _e83ac8c_

I was able to utilize the new keycloak oidc backend and the old too.
